.ci/test.sh

```diff
@@ -195,40 +195,48 @@ if [[ $TASK == "gpu" ]]; then
     elif [[ $METHOD == "source" ]]; then
         cmake -DUSE_GPU=ON ..
     fi
-elif [[ $TASK == "cuda" || $TASK == "cuda_exp" ]]; then
-    if [[ $TASK == "cuda" ]]; then
+elif [[ $TASK == "cuda" ]]; then
+    if [[ $DEVICE == "cuda" ]]; then
         sed -i'.bak' 's/std::string device_type = "cpu";/std::string device_type = "cuda";/' $BUILD_DIRECTORY/include/LightGBM/config.h
         grep -q 'std::string device_type = "cuda"' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
-    else
+    elif [[ $DEVICE == "cuda_exp" ]]; then
         sed -i'.bak' 's/std::string device_type = "cpu";/std::string device_type = "cuda_exp";/' $BUILD_DIRECTORY/include/LightGBM/config.h
         grep -q 'std::string device_type = "cuda_exp"' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
         # by default ``gpu_use_dp=false`` for efficiency. change to ``true`` here for exact results in ci tests
         sed -i'.bak' 's/gpu_use_dp = false;/gpu_use_dp = true;/' $BUILD_DIRECTORY/include/LightGBM/config.h
         grep -q 'gpu_use_dp = true' $BUILD_DIRECTORY/include/LightGBM/config.h || exit -1  # make sure that changes were really done
+    else
+        exit -1
```

Comment:
```suggestion
        echo "did not recognize DEVICE '${DEVICE}'. Should be either 'cuda' or 'cuda_exp'."
        exit -1
```

Can you please add a log line here explaining what happened? Otherwise, the job will just fail and it won't be clear why or that it happened because of this specific line.

Reply:
Since now we don't introduce `device` and `DEVICE`, I've removed these changes in `.ci/test.sh`.

Reply:
agree, thanks